### PR TITLE
allow id>=0 for the first particle in chain analysis routines

### PR DESCRIPTION
**Pull request by stuehn**
Chain analysis objects in ESPResSo++ (e.g. MeanSquareInternalDist) only work with monodisperse chainlengths systems where all the particles have consecutively numbered ids. In ES++ versions < 2.0 the starting pid had to be 0. In newer versions this had been changed to 1 which led to some confusion. I added a start_pid parameter (default is 0) to the MSID object to allow for starting pids >=0.

### CODE REVIEW

File: src/analysis/ConfigsParticleDecomp.hpp [hunk 1]
```diff
@@ -147,13 +147,14 @@ namespace espressopp {
...
        */
-      ConfigsParticleDecomp(shared_ptr<System> system, int _chainlength): SystemAccess (system){
+      ConfigsParticleDecomp(shared_ptr<System> system, int _chainlength, int _start_pid): SystemAccess (system){
         // by default key = "position", it will store the particle positions
         // (option: "velocity" or "unfolded")
         esutil::Error err(system->comm);
         
         key = "position";
         chainlength = _chainlength;
+	start_pid = _start_pid;
```
Comment (junghans) [inline src/analysis/ConfigsParticleDecomp.hpp]: Indentation?

File: src/analysis/MeanSquareDispl.py [hunk 2]
```diff
@@ -29,8 +29,10 @@
 
 		:param system:
 		:param chainlength: (default: None)
+                :param start_pid: (default: 0)
```
Comment (junghans) [inline src/analysis/MeanSquareDispl.py]: Indentation?

File: src/analysis/MeanSquareDispl.py [hunk 3]
```diff
@@ -29,8 +29,10 @@
...
 		:type system:
 		:type chainlength:
+                :type start_pid:
```
Comment (junghans) [inline src/analysis/MeanSquareDispl.py]: Indentation?

File: src/analysis/ConfigsParticleDecomp.hpp [hunk 4]
```diff
@@ -147,13 +147,14 @@ namespace espressopp {
         * !! currently only works for particles numbered like 0, 1, 2,... !!
         * !! with each chain consisting particles with subsequent ids     !!
        */
-      ConfigsParticleDecomp(shared_ptr<System> system, int _chainlength): SystemAccess (system){
+      ConfigsParticleDecomp(shared_ptr<System> system, int _chainlength, int _start_pid): SystemAccess (system){
```
Comment (jkrajniak) [inline src/analysis/ConfigsParticleDecomp.hpp]: Please update the comment above.

File: src/analysis/ConfigsParticleDecomp.py [hunk 5]
```diff
@@ -30,6 +30,15 @@
 		:param system:
 		:type system:
 
+.. function:: espressopp.analysis.ConfigsParticleDecomp(system, chainlength, start_pid)
+
+		:param system:
+                :param chainlength:
+                :param start_pid:
+		:type system:
```
Comment (jkrajniak) [inline src/analysis/ConfigsParticleDecomp.py]: indentation

File: src/analysis/ConfigsParticleDecomp.hpp (hunk 1 above)
Comment (stuehn) [reply src/analysis/ConfigsParticleDecomp.hpp]: strange ... my editor somehow put <tabs> in the beginning ... I resolved the problem for all subsequent indentations

**Comment by stuehn** [comment]
all requests have been resolved

**Pull request merged**